File: backend/insurance_engine/src/InsuranceEngine.Policy/Domain/ValueObjects/Applicant.cs
using System;
using System.Collections.Generic;

namespace InsuranceEngine.Policy.Domain.ValueObjects;

/// <summary>
/// Applicant/Proposer information. Stored as JSONB in policies table.
/// </summary>
public record Applicant
{
    public string FullName { get; init; } = string.Empty;
    public DateTime? DateOfBirth { get; init; }
    public string? NidNumber { get; init; }      // Encrypted at rest
    public string? Occupation { get; init; }
    public long AnnualIncome { get; init; }       // paisa
    public string? Address { get; init; }
    public string? PhoneNumber { get; init; }     // Encrypted at rest
    public HealthDeclaration? HealthDeclaration { get; init; }
}

/// <summary>
/// Health declaration information. Stored as JSONB.
/// </summary>
public record HealthDeclaration
{
    public bool HasPreExistingConditions { get; init; }
    public List<string> Conditions { get; init; } = new();
    public bool IsSmoker { get; init; }
    public string? BloodGroup { get; init; }
}

File: sdks/insuretech-go-sdk/pkg/models/applicant.go
package models

import (
	"time"
)

// Applicant represents a applicant
type Applicant struct {
	Address string `json:"address,omitempty"`
	AnnualIncome *Money `json:"annual_income,omitempty"`
	DateOfBirth time.Time `json:"date_of_birth,omitempty"`
	FullName string `json:"full_name,omitempty"`
	HealthDeclaration *PolicyHealthDeclaration `json:"health_declaration,omitempty"`
	NidNumber string `json:"nid_number,omitempty"`
	Occupation string `json:"occupation,omitempty"`
}
